pkg/skaffold/deploy/util.go

```diff
@@ -18,45 +18,30 @@ package deploy
 
 import (
 	"bufio"
-	"fmt"
+	"bytes"
 	"io"
 
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl"
 	"github.com/sirupsen/logrus"
 
 	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
-	"k8s.io/client-go/kubernetes/scheme"
 )
 
-func parseRuntimeObject(namespace string, b []byte) (Artifact, error) {
-	d := scheme.Codecs.UniversalDeserializer()
-	obj, _, err := d.Decode(b, nil, nil)
-	if err != nil {
-		return Artifact{}, fmt.Errorf("error decoding parsed yaml: %s", err.Error())
-	}
-	return Artifact{
-		Obj:       obj,
-		Namespace: namespace,
-	}, nil
-}
-
-func parseReleaseInfo(namespace string, b *bufio.Reader) []Artifact {
-	results := []Artifact{}
+func parseReleaseInfo(b *bufio.Reader) kubectl.ManifestList {
 	r := k8syaml.NewYAMLReader(b)
+	var manifests kubectl.ManifestList
 	for {
-		doc, err := r.Read()
+		b, err := r.Read()
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
-			logrus.Infof("error parsing object from string: %s", err.Error())
+			logrus.Infof("error reading line string: %s", err.Error())
 			continue
 		}
-		obj, err := parseRuntimeObject(namespace, doc)
-		if err != nil {
-			logrus.Infof(err.Error())
-		} else {
-			results = append(results, obj)
+		if i := bytes.Index(b, []byte("apiVersion")); i >= 0 {
```

Comment:
This is a bug, I think. It currently works, because in all examples you put `apiVersion` at the very beginning. But this is also a perfectly valid k8s yaml resource:
```yaml
kind: Pod
metadata:
  name: pod
  labels:
    app: skaffold-helm
spec:
  container:
    image: some
apiVersion: v1
```
But the appended manifest would just be `apiVersion: v1`, instead of the full yaml.

Reply:
This is actually parsing the output of `helm get manifest` we use the trust like we do in `kubectl get`
1.  "---" is the yaml separtor
2. we can rely in apiVersion being present at the top.

Reply:
Alright, I'm convinced.